src/test/java/com/smartbear/readyapi/client/DataGenDataSourceTest.java

```diff
@@ -0,0 +1,79 @@
+package com.smartbear.readyapi.client;
+
+import com.smartbear.readyapi.client.model.BooleanDataGenerator;
+import com.smartbear.readyapi.client.model.DataGenDataSource;
+import com.smartbear.readyapi.client.model.DataSourceTestStep;
+import com.smartbear.readyapi.client.model.TestStep;
+import com.smartbear.readyapi.client.teststeps.TestStepTypes;
+import org.junit.Test;
+
+import static com.smartbear.readyapi.client.TestRecipeBuilder.newTestRecipe;
+import static com.smartbear.readyapi.client.teststeps.TestSteps.dataGenDataSource;
+import static com.smartbear.readyapi.client.teststeps.datasource.datagen.DataGenerators.booleanTypeProperty;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public class DataGenDataSourceTest {
+
+    @Test
+    public void buildsRecipeWithDataSourceTestStepWithBooleanDataGenDataSource() throws Exception {
+        TestRecipe recipe = newTestRecipe()
+                .addStep(dataGenDataSource()
+                        .withNumberOfRows(34)
+                        .addDataGenerator(
```

Comment:
hmm... could addDataGenerator and booleanTypeProperty be merged into one method? Something like

```
dataGenDataSource()
.withNumberOfRows(34)
.withBooleanTypeProperty("property1")
   .duplicatedBy( 1 )
   .withYesNoFormat()
.andCityTypeProperty("city1")
  ....
```
or would that be overdoing the fluent thing? @spindelmanne thoughts?